File: EnforcerWeb/src/app/model/Artifact.ts
export class Artifact {
	public artifact_id: number;
	public status: string;
	public group_name: string;
	public artifact_name: string;
	public version_name: string;
	public is_vendor_licensed: boolean;
	public created_date: Date;
	public approval_date?: Date;
	public approval_authorization?: string;
	public approval_ts?: Date;
	public deprecation_authorization?: string;
	public deprecation_date?: Date;
	public deprecation_ts?: Date;
	public retirement_authorization?: string;
	public retirement_date?: Date;
	public retirement_ts?: Date;

	constructor( obj: Object = {}) {
		Object.assign(this, obj) ;
	}
 
	public toJSON() {
		return {
			artifact_id: this.artifact_id,
			status: this.status,
			group_name: this.group_name,
			artifact_name: this.artifact_name,
			version_name: this.version_name,
			is_vendor_licensed: this.is_vendor_licensed,
			created_date: this.created_date,
			approval_date: this.approval_date,
			approval_authorization: this.approval_authorization,
			approval_ts: this.approval_ts,
			deprecation_authorization: this.deprecation_authorization,
			deprecation_date: this.deprecation_date,
			deprecation_ts: this.deprecation_ts,
			retirement_authorization: this.approval_authorization,
			retirement_date: this.retirement_date,
			retirement_ts: this.retirement_ts,
		}
	}
}

File: EnforcerServices/src/main/java/com/webforged/enforcer/management/data/Artifact.java
package com.webforged.enforcer.management.data;

import java.time.Instant;
import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
@Table("artifacts")
public class Artifact {
	
	@Id
	public Long artifact_id ;
	private String status;
	private String group_name;
	private String artifact_name;
	private String version_name;
	private Boolean is_vendor_licensed;
	private LocalDateTime created_date;
	private LocalDateTime approval_date;
	private String approval_authorization;
	private Instant approval_ts;
	private LocalDateTime deprecation_date;
	private String deprecation_authorization;
	private Instant deprecation_ts;
	private LocalDateTime retirement_date;
	private String retirement_authorization;
	private Instant retirement_ts;
	
	public Artifact() {
	}
}
